x/posts/internal/keeper/keeper.go

```diff
@@ -139,7 +152,23 @@ func (k Keeper) GetPostsFiltered(ctx sdk.Context, params types.QueryPostsParams)
 			matchCreator = params.Creator.Equals(post.Creator)
 		}
 
-		if matchParentID && matchCreationTime && matchAllowsComments && matchSubspace && matchCreator {
+		// match hashtags if provided
+		if len(params.Hashtags) > 0 {
+			postHashtags := k.GetPostHashtags(post)
+			if len(postHashtags) == len(params.Hashtags) {
+				sort.Strings(postHashtags)
+				sort.Strings(params.Hashtags)
+				for index, hashtag := range params.Hashtags {
+					if matchHashtags = postHashtags[index] == hashtag; !matchHashtags {
+						break
+					}
+				}
+			} else {
+				matchHashtags = false
+			}
```

Comment:
I think that this piece of code can be simplified using the `&&` operators for bool values and remembering that: 

- `true && false` = `false`
- `true && true` = `true`

```suggestion
			postHashtags := k.GetPostHashtags(post)
			matchHashtags = len(postHashtags) == len(params.Hashtags)
			
			sort.Strings(postHashtags)
			sort.Strings(params.Hashtags)
			for index, hashtag := range params.Hashtags {
				matchHashtags = postHashtags[index] == hashtag
			}
```

Considering that the hashtags number will never be _that_ large, I think that iterating over their values even if the length is different does not degrade the overall performance in a significant way.

Reply:
I reviewed your solution as it got some problems and committed it in the latest commit